test/e2e/util.go

```diff
@@ -1268,6 +1268,21 @@ func waitForRCPodsRunning(c *client.Client, ns, rcName string) error {
 	return nil
 }
 
+// Wait up to 10 minutes for pods to become Running.
+func waitForPodsWithLabel(c *client.Client, ns string, label labels.Selector) (pods *api.PodList, err error) {
+	for t := time.Now(); time.Since(t) < podListTimeout; time.Sleep(poll) {
+		pods, err = c.Pods(ns).List(label, fields.Everything())
```

Comment:
I think I found the cause. Getting pods back based on label may fail due to timing issue. Therefore, I modified this `waitForPodsWithLabel()` to have it keep polling pods with label `run=<rc name>` for a certain time until it gets more than 0 pods. This seems to work and I'll test again just to make sure. Thanks for your help!